jtrust-lib/src/main/java/be/fedict/trust/crl/OnlineCrlRepository.java

```diff
@@ -105,7 +106,10 @@ private X509CRL getCrl(final URI crlUri) throws IOException,
 
 		if (null != this.networkConfig) {
 			final HttpHost proxy = new HttpHost(this.networkConfig.getProxyHost(), this.networkConfig.getProxyPort());
+			int timeout = 5; // seconds
 			httpClient.getParams().setParameter(ConnRoutePNames.DEFAULT_PROXY, proxy);
+			httpClient.getParams().setParameter(CoreConnectionPNames.CONNECTION_TIMEOUT, timeout * 1000);
+			httpClient.getParams().setParameter(CoreConnectionPNames.SO_TIMEOUT, timeout * 1000);
```

Comment:
once you get a connection, you may still have to wait for the data to be returned, which could be another 2 seconds